service/src/java/org/apache/hive/service/cli/thrift/ThriftCLIService.java

```diff
@@ -165,7 +165,7 @@ public ThriftCLIService(CLIService service, String serviceName) {
   public synchronized void init(HiveConf hiveConf) {
     this.hiveConf = hiveConf;
 
-    String hiveHost = System.getenv("HIVE_SERVER2_THRIFT_BIND_HOST");
```

Comment:
This shadowed the `hiveHost` class variable at line 131 (  `protected String hiveHost;` ). This meant all assignments to `hiveHost` in `init` were to the local copy of `hiveHost`, instead of the class variable which is used by most other code (including classes that subclass this one), which is always null due to the shadowing here.